src/ble/hal/utils/statemachine.cpp

```diff
@@ -292,11 +294,13 @@ int StateMachine::shouldMoveState(Event::Type eventType) const
 
 static gboolean timerEvent(gpointer user_data)
 {
-    StateMachine *sm = (StateMachine*)user_data;
-    if (sm == nullptr || !sm->isRunning()) {
+    StateMachine_userdata *userData = (StateMachine_userdata*)user_data;
+    if (userData == nullptr || !userData->m_ptr->isRunning() || !userData->m_isAlive || !(*userData->m_isAlive)) {
```

Comment:
shouldn't this be at the end of the "if" so that isAlive is checked before dereferencing the pointer?
```suggestion
    if (userData == nullptr || !userData->m_isAlive || !(*userData->m_isAlive) || !userData->m_ptr->isRunning()) {
```